### ISSUE TITLE
Bloated testing output due to deprecated warnings

### ISSUE THREAD

**Issue by b-butler** [glotzerlab/signac-flow#244]
### Description
Reliance on deprecated functions, methods, etc. needs to be removed from out testing. 

### To reproduce

Just run the flow testing suite.

### Fix

Go through the testing output and find all references to deprecated functions and remove them.

**Comment by csadorf** [comment]
Thank you for creating this issue. I specifically asked to split fixing this from the test refactoring PR, because I would like that change to be 100% refactoring, no other changes.

I think we should absolutely refactor the package code to stop relying on deprecated functions internally, which should already reduce the noise, however I'd like to be a bit careful with updating all the tests, and rather mark them as explicitly testing deprecated API and suppress warnings instead of changing them all. Unit tests that test deprecated API and thus trigger deprecation warnings are vital to actually ensure that deprecated behavior is still supported and thus actually maintaining backwards compatibility.

Deprecated API that should be removed should of course lead to failed tests.

**Comment by b-butler** [comment]
Many tests that aren't designed explicitly to test deprecated API are uses deprecated functions. I do see your point with maintaining backwards compatible with deprecated code until the correct version. I do think however that a few contained test explicitly for deprecated code with the main code base relying on non-deprecated code would be preferable. I do believe for the most part the package does not call deprecated functions.

**Comment by kidrahahjo** [comment]
Can I work on removing the deprecated warnings? While working for pytest implementation, this issue occurred for me too, and it became frustrating scrolling to the top to see what errors I'm getting.

**Comment by csadorf** [comment]
@kidrahahjo Yes, of course! Let's just make sure that #232 is merged first.

**Comment by bdice** [comment]
@kidrahahjo This should be unblocked. Thanks for offering to help, and let us know if you have questions! You'll want to start from the current `glotzerlab:master` branch (not the branch you created for the previous PR).

**Comment by kidrahahjo** [comment]
> @kidrahahjo This should be unblocked. Thanks for offering to help, and let us know if you have questions! You'll want to start from the current `glotzerlab:master` branch (not the branch you created for the previous PR).

Okay, thanks. 
I have started working on this. Will raise a PR soon.

**Comment by kidrahahjo** [comment]
@bdice @csadorf 
To fix the deprecation warning of get_statepoint, I'd need to commit to the signac repository too. Should I do that?

**Comment by csadorf** [comment]
@kidrahahjo Yes, please create a separate PR for that.

**Comment by bdice** [comment]
@csadorf: @kidrahahjo and I discussed on https://github.com/glotzerlab/signac/pull/286 and it is not likely that changes will be needed in the signac repository.

@kidrahahjo For warnings about `get_id()`, it will help you to know that the `id` property it recommends switching to was only added recently. For testing with signac 1.0.0, we still need to leave the test as using `get_id()`. I recommend that you use a warnings filter to hide warnings that cannot or should not be removed (because they are testing deprecated features that have not yet been removed). https://docs.pytest.org/en/latest/warnings.html

The best thing to do in this pull request is to identify the source of those warnings, and handle them in one of two ways:
1. If the warning must be raised for backwards compatibility with older versions of signac (like `get_id()`), please add warning filters to those tests.
2. If the warning is raised by code inside of signac-flow that could be updated or turned into a private method (like what we did with `get_statepoint` in https://github.com/glotzerlab/signac/pull/282), then please list those warnings and the functions causing them in a comment on this issue. I would like to offer specific direction on how to handle each one individually.

Some deprecations are for functionality that we need internally (e.g. as a private method) but do not want to expose to users as public methods. Some deprecations are for features that we intend to remove completely in a future release. Since you may not be familiar with the project history, I would like to offer specific guidance on each one.

**Comment by kidrahahjo** [comment]
I'll list all the deprecation errors.
Please let me know whether I should go for a warning filter or turn it into a private method.

1. get_id() deprecation - I'll go for warning filter
2. sys.getcheckinterval() and sys.setcheckinterval() are deprecated
3. always is deprecated as of 0.9 and will be removed in 0.11

**Comment by bdice** [comment]
@kidrahahjo Your assessment is correct, just need more details on (2):
1. `get_id()`, use a warning filter.
2. `sys.getcheckinterval()` -- I couldn't find this in the codebase. Where is this causing a warning? Maybe it is in a dependency of signac-flow. More information would be helpful here.
3. `always`, use a warning filter.
4. `fractions.gcd` -- [replace with `math.gcd`](https://docs.python.org/3.8/library/fractions.html#fractions.gcd).
5. `warn` method -- yes, replace it with `logger.warning`.

**Comment by kidrahahjo** [comment]
> @kidrahahjo Your assessment is correct, just need more details on (2):
> 
> 1. `get_id()`, use a warning filter.
> 2. `sys.getcheckinterval()` -- I couldn't find this in the codebase. Where is this causing a warning? Maybe it is in a dependency of signac-flow. More information would be helpful here.
> 3. `always`, use a warning filter.
> 4. `fractions.gcd` -- [replace with `math.gcd`](https://docs.python.org/3.8/library/fractions.html#fractions.gcd).
> 5. `warn` method -- [replace with `logger.warning`](https://docs.python.org/3/library/logging.html#logging.Logger.warning).

sys.getcheckinterval() will be found in 65: flow/util/tqdm/_tqdm.py

**Comment by bdice** [comment]
@kidrahahjo Let's skip the tqdm warning in this pull request. We would need to upgrade our vendored tqdm module to fix that, or switch to using tqdm as a requirement instead of a vendored package. I will open a separate issue for that discussion.

**Comment by kidrahahjo** [comment]
> @kidrahahjo Let's skip the tqdm warning in this pull request. We would need to upgrade our vendored tqdm module to fix that, or switch to using tqdm as a requirement instead of a vendored package. I will open a separate issue for that discussion.

@bdice Okay, then I'll do the changes and raise a PR
